var/spack/repos/builtin/packages/sherpa/package.py

```diff
@@ -100,30 +111,48 @@ def configure_args(self):
-            args.append('--enable-fastjet=' + self.spec['fastjet'].prefix)
-        if self.spec.satisfies('+openloops'):
-            args.append('--enable-openloops=' + self.spec['openloops'].prefix)
-        if self.spec.satisfies('+recola'):
-            args.append('--enable-recola=' + self.spec['recola'].prefix)
-        if self.spec.satisfies('+root'):
-            args.append('--enable-root=' + self.spec['root'].prefix)
-        if self.spec.satisfies('+lhapdf'):
-            args.append('--enable-lhapdf=' + self.spec['lhapdf'].prefix)
-        if self.spec.satisfies('+hztool'):
-            args.append('--enable-hztool=' + self.spec['hztool'].prefix)
-        if self.spec.satisfies('+cernlib'):
-            args.append('--enable-cernlib=' + self.spec['cernlib'].prefix)
+        hepmc_root = lambda x: self.spec['hepmc'].prefix
+        args.extend(self.enable_or_disable('hepmc2', activation_value=hepmc_root))
+        if self.spec.satisfies('@2.2.13:'):
+            args.extend(self.enable_or_disable('hepmc3', activation_value='prefix'))
+            args.extend(self.enable_or_disable('rivet', activation_value='prefix'))
+            args.extend(self.enable_or_disable('lhapdf', activation_value='prefix'))
+        else:
+            # See https://gitlab.com/sherpa-team/sherpa/-/issues/348
+            if self.spec.satisfies('+hepmc3'):
+                args.append('--enable-hepmc3=' + self.spec['hepmc3'].prefix)
+            if self.spec.satisfies('+rivet'):
+                args.append('--enable-rivet=' + self.spec['rivet'].prefix)
+            if self.spec.satisfies('+lhapdf'):
+                args.append('--enable-lhapdf=' + self.spec['lhapdf'].prefix)
+
+        args.extend(self.enable_or_disable('fastjet', activation_value='prefix'))
+        args.extend(self.enable_or_disable('openloops', activation_value='prefix'))
+        args.extend(self.enable_or_disable('recola', activation_value='prefix'))
+        args.extend(self.enable_or_disable('root', activation_value='prefix'))
+
+        # args.extend(self.enable_or_disable('hztool', activation_value='prefix'))
+        # args.extend(self.enable_or_disable('cernlib', activation_value='prefix'))
+        args.extend(self.enable_or_disable('blackhat', activation_value='prefix'))
+        args.extend(self.enable_or_disable('ufo'))
+
+        if self.spec.satisfies('+mpi'):
+            args.append('CC=' + self.spec['mpi'].mpicc)
+            args.append('MPICXX=' + self.spec['mpi'].mpicxx)
+            args.append('CXX=' + self.spec['mpi'].mpicxx)
+            args.append('FC=' + self.spec['mpi'].mpifc)
+
         return args
+
+    def flag_handler(self, name, flags):
+        flags = list(flags)
+        mflag = '-m64' if platform.machine() == 'x86_64' else ''
+        if '+cms' in self.spec:
+            if name == 'cxxflags':
+                flags.extend(['-fuse-cxa-atexit', mflag, '-O2', '-std=c++' + self.spec.variants['cxxstd'].value])
```

Comment:
Use the information available in the spec:
```suggestion
        if '+cms' in self.spec:
            if name == 'cxxflags':
                flags.extend(['-fuse-cxa-atexit', '-O2', '-std=c++' + self.spec.variants['cxxstd'].value])
            if self.spec.target.family == 'x86_64':
                flags.append('-m64')
```
And drop the `import platform` at the top.